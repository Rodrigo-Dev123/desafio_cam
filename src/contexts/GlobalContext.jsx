import P from 'prop-types';
import { createContext, useState, useEffect } from "react";

export const GlobalContext = createContext();

const GlobalProvider = ({ children }) => {
    const [items, setItems] = useState([]);
    const [itemsPerPage] = useState(10);
    const [currentPage, setCurrentPage] = useState(0);

    const [modal, setModal] = useState(false);
    const [postModal, setPostModal] = useState(null);

    const startIndex = currentPage * itemsPerPage;
    const endIndex = startIndex + itemsPerPage;
    const pages = Math.ceil(items.length / itemsPerPage);
    const currentItems = items.slice(startIndex, endIndex);

    const removeItem = (id) => {
        const updatedItems = JSON.parse(localStorage.getItem('items')).filter(item => {
            return item.id !== id;
        })

        localStorage.setItem('items', JSON.stringify(updatedItems))
        setItems(JSON.parse(localStorage.getItem('items')));
        alert(`Foto ${id} apagada com sucesso! ｡◕‿◕｡`);
    }

    useEffect(() => {
        if (!localStorage.getItem('items')) {
            const fetchData = async () => {
                console.log('fetch execultou!');
                const result = await fetch('https://jsonplaceholder.typicode.com/photos')
                    .then(response => response.json())
                    .then(data => data)

                localStorage.setItem('items', JSON.stringify(result));
                setItems(result);
            }

            fetchData();
            return;
        }
        setItems(JSON.parse(localStorage.getItem('items')));
    }, []);

    return (
        <GlobalContext.Provider value={{
            currentItems, items, setItems, itemsPerPage,
            currentPage, setCurrentPage, modal, setModal,
            startIndex, endIndex, pages, removeItem,
            postModal, setPostModal
        }}>
            {children}
        </GlobalContext.Provider>
    )
}

GlobalProvider.propTypes = {
    children: P.node.isRequired,
};

export default GlobalProvider;
