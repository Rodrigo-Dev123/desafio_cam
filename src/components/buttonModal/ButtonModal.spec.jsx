import { fireEvent, render, screen } from '@testing-library/react';
import { GlobalContext } from '../../contexts/GlobalContext';
import ButtonModal from './index';
import userEvent from '@testing-library/user-event';

const contextValue = {
    setPostModal: jest.fn(),
    modal: jest.fn(),
    setModal: jest.fn(),
    currentItems: [
        {
            "albumId": 1,
            "id": 1,
            "title": "accusamus beatae ad facilis cum similique qui sunt",
            "url": "https://via.placeholder.com/600/92c952",
            "thumbnailUrl": "https://via.placeholder.com/150/92c952"
        },
        {
            "albumId": 1,
            "id": 2,
            "title": "reprehenderit est deserunt velit ipsam",
            "url": "https://via.placeholder.com/600/771796",
            "thumbnailUrl": "https://via.placeholder.com/150/771796"
        },
        {
            "albumId": 1,
            "id": 3,
            "title": "officia porro iure quia iusto qui ipsa ut modi",
            "url": "https://via.placeholder.com/600/24f355",
            "thumbnailUrl": "https://via.placeholder.com/150/24f355"
        },
    ],
};

describe('<ButtonModal />', () => {
    it('Should render ButtonModal', () => {
        const { debug } = render(
            <GlobalContext.Provider value={contextValue}>
                <ButtonModal id={'buttonModal'}/>
            </GlobalContext.Provider>
        );
        debug();
        expect(screen.getByRole('button', {name: /Ver mais/i})).toBeInTheDocument();
    });

    it('Shouls math snapshot', () => {
        const { container } = render(
            <GlobalContext.Provider value={contextValue}>
                <ButtonModal />
            </GlobalContext.Provider>
        );

       expect(container.firstChild).toMatchSnapshot();
    });
});