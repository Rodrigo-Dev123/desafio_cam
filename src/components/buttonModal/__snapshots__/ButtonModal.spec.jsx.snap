// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonModal /> Shouls math snapshot 1`] = `
<input
  type="button"
  value="Ver Mais"
/>
`;
